augurpy/read_load.py

```diff
@@ -16,21 +22,53 @@ def load(input: Union[AnnData, DataFrame], meta: DataFrame, label_col: str, cell
         cell_type_col: column of the meta DataFrame or the Anndata or matrix containing the cell type labels for each
             cell in the cell-by-gene expression matrix
 
-
     Returns:
-        Pandas DataFrame containing gene expression values (cells in rows, genes in columns), cell type and condition
+        Anndata object containing gene expression values (cells in rows, genes in columns) and cell type, label as obs
     """
-    pass
+    if isinstance(input, AnnData):
+        input.obs = input.obs.rename(columns={cell_type_col: "cell_type", label_col: "label"})
+        out = input
+
+    elif isinstance(input, DataFrame):
+        try:
+            cell_type = input[cell_type_col]
+            label = input[label_col]
+        except KeyError:
+            print("No column names matching cell_type_col and label_col. Looking in meta data.")
+            try:
+                cell_type = meta[cell_type_col]
+                label = meta[label_col]
+            except (KeyError, TypeError):
+                raise Exception("Missing label and / or cell type column.") from None
+
+            else:
+                print("Adding cell type and label from metadata.")
```

Comment:
Not sure whether this status message is necessary.

Reply:
Yeah your right. Removing it